src/MergeInfillLines.cpp

```diff
@@ -248,9 +249,19 @@ MergeInfillLines::MergeInfillLines(ExtruderPlan& plan)
 
             if (second_path.config->isTravelPath())
             {
+                has_first_path = false;
                 continue; //Skip travel paths, we're looking for the first non-travel path.
             }
 
+            // Use the first non-travel path as the first path that can be used for merging. After we encounter
+            // a travel path, we need to find another first non-travel path for merging.
+            if (!has_first_path) {
```

Comment:
Code style: Brackets on new line.
```suggestion
            if (!has_first_path)
            {
```